clients/unshaded/src/main/java/tachyon/client/InStream.java

```diff
@@ -15,57 +15,21 @@
 
 package tachyon.client;
 
-import java.io.InputStream;
 import java.io.IOException;
+import java.io.InputStream;
 
-import tachyon.conf.TachyonConf;
-
-/**
- * <code>InStream</code> is the base class of input stream for TachyonFile streaming input. To get
- * an instance of this class, one should call the static method <code>getInStream</code> of
- * <code>tachyon.client.TachyonFile</code>, rather than constructing a new instance directly in the
- * client code.
- */
 public abstract class InStream extends InputStream {
-  protected final TachyonFile mFile;
-  protected final TachyonFS mTachyonFS;
-  protected final ReadType mReadType;
-  protected final TachyonConf mTachyonConf;
-
   /**
-   * @param file the input file of the InStream
-   * @param readType the InStream's read type
-   * @param tachyonConf the TachyonConf instance for this stream.
+   * Gets the remaining number of bytes left in the stream, starting at the current position.
```

Comment:
Instead of having a `InStream` abstract class into the inheriting hierarchy, why not making this an interface and making its subclasses simply extending the interface? In this way, we don't need this extra level of inheritance. 

Similar comment applies to abstract class `OutStream`.